apps/condo/domains/billing/schema/resolvers/propertyResolver.js

```diff
@@ -27,21 +26,43 @@ class PropertyResolver extends Resolver {
     constructor ({ billingContext, context }) {
         super(billingContext, context, { name: 'property' })
         this.organizationProperties = []
-        this.properties = []
         this.transform = new AddressTransform()
         this.propertyFinder = new PropertyFinder()
         this.parser = new AddressFromStringParser()
         this.tin = get(billingContext, 'organization.tin')
+        this.organizationId = get(billingContext, 'organization.id')
         this.transform.init(get(billingContext, 'settings.addressTransform', {}))
         this.isCottageVillage = !!(get(billingContext, 'settings.isCottageVillage'))
         this.addressService = createInstance()
     }
 
+    async getBillingProperty (where = {}) {
+        const [property] = await find('BillingProperty', { ...where, context: { id: this.billingContext.id }, deletedAt: null })
+        return property
+    }
+
+    async getOrganizationProperty (addressKey) {
+        if (!this.organizationId) {
+            return null
+        }
+        const loaded = this.organizationProperties.find(({ addressKey: propertyAddressKey }) => propertyAddressKey === addressKey)
+        if (loaded) {
+            return loaded
+        }
+        const [property] = await find('Property', { addressKey, organization: { id: this.organizationId }, deletedAt: null })
+        if (property) {
+            this.organizationProperties.push(property)
+        }
+        return property
+    }
+
     async init () {
-        const organizationProperties = await find('Property', { organization: { id: get(this.billingContext, 'organization.id') }, deletedAt: null })
-        this.organizationProperties = Object.fromEntries(organizationProperties.map(({ addressKey, address }) => ([addressKey, address])))
-        this.properties = await find('BillingProperty', { context: { id: this.billingContext.id }, deletedAt: null })
-        this.propertyFinder.init(organizationProperties)
+        if (this.isCottageVillage) {
+            // We still need to load all properties in case we do not use addressService for address normalize
+            const organizationProperties = await find('Property', { organization: { id: this.organizationId }, deletedAt: null })
```

Comment:
```suggestion
            const organizationProperties = await find('Property', { organization: { id: this.organizationId, deletedAt: null }, deletedAt: null })
```

Reply:
We have organization and context as an input, so we already check that organization and context are not deleted
properties, accounts, receipts - are checked to not to be deleted